### ISSUE TITLE
Accuracy issue in torch dynamo

### ISSUE THREAD

**Issue by naveenkumarmarri** [pytorch/pytorch#153299]
### 🐛 Describe the bug

I am noticing accuracy difference when training with torch compile. to narrow down the issue, ran ablation with `eager`, `aot_eager` and `inductor` and observed that numerics diverge when using backend=`inductor`.

Looks like this is happening inside dynamo and to produce, I added `repro.py` and `minified_launcher.py`. let me know if you need any information that can help better reproduce this 

### Versions

ran repro minifer and the script is available here
repro.py - https://gist.github.com/naveenkumarmarri/0f66d89695e56840a06c7a37dccca83f
minifier.py - https://gist.github.com/naveenkumarmarri/8fa35e72e3210a6c6b13548d9ab73df6

**Comment by masnesral** [comment]
@naveenkumarmarri can you also try with `torch._inductor.config.emulate_precision_casts=True`

**Comment by naveenkumarmarri** [comment]
@masnesral adding the flag still introduces the accuracy gap

**Comment by masnesral** [comment]
@naveenkumarmarri does the repro script actually work for you? For example: https://gist.github.com/naveenkumarmarri/0f66d89695e56840a06c7a37dccca83f#file-repro-py-L67

**Comment by navmarri14** [comment]
this script is autogenerated by enabling
```
export TORCHDYNAMO_REPRO_AFTER=aot
export TORCHDYNAMO_REPRO_LEVEL=4
```
I followed the torch.compile [docs](https://docs.pytorch.org/docs/stable/torch.compiler_troubleshooting.html#minifier) to get the script. Let me know if there is a better way to generate the repro script that might be helpful to reproducing this

**Comment by masnesral** [comment]
@navmarri14, I dunno; I'm kind of a noob with minifier support, but the script looks invalid to me. Is this use case using either of `post_grad_custom_pre_pass` or `post_grad_custom_post_pass` per chance? I don't think the minifier works if you're using those customizations.

**Comment by navmarri14** [comment]
@masnesral I don't have `post_grad_custom_pre_pass` or `post_grad_custom_post_pass` as part of the model. I was able to narrow down the issue to the following operation. 

```python3
# compiled model leads to mismatch in accuracy
def fn(x): 
  return torch.logsumexp(x, dim=-1).pow(2).mean()
```

if I disable the compilation on the specific operation, the results match with the uncompiled model.
```python3
@torch._dynamo.disable(recursive=True)
def fn(x): 
  return torch.logsumexp(x, dim=-1).pow(2).mean()
```

I tried to reproduce this in a simple model definition but the results seem to be matching between compiled and uncompiled models. 

```python3
import torch
import torch.nn as nn

class ToyModel(nn.Module):
    def __init__(self):
        super(ToyModel, self).__init__()
        
    def forward(self, x):
        return torch.logsumexp(x, dim=-1).pow(2).mean()


device = torch.device("cuda")
model = ToyModel().to(device)
x = torch.load("shift_logits.pt", map_location=device)
uncompiled_loss = model(x)
print(uncompiled_loss)

model = torch.compile(model, fullgraph=False, backend="inductor")
compiled_loss = model(x)
print(compiled_loss)
print(torch.equal(uncompiled_loss, compiled_loss))
```

cc: @ezyang incase if you have seen this issue before or if could share better approach to share repro

**Comment by navmarri14** [comment]
@masnesral are there any references on how to fix these kind of issues. I can make a PR if possible

**Comment by eellison** [comment]
I started looking at this, i can take it.

**Comment by mlazos** [comment]
@naveenkumarmarri can you share more of the model code if possible? 

I couldn't find a section of code in the repro that corresponds to logsumexp, since I didn't see `exp` anywhere. 

Are you able to record the exact inputs to that function and share them?

**Comment by mlazos** [comment]
Offline during inductor debug session:
Will try removing the downcasts to bfloat16 and try at higher precision
Disable pattern matcher
